action.yml

```diff
@@ -8,7 +8,14 @@ branding:
 
 inputs:
   scheme:
-    description: 'Versioning scheme - semver, or, calver (defaults to semver)'
+    description: |
+      Versioning scheme - semver, calver or conventional_commits (defaults to semver).
+
+      `conventional_commits` Will parse the last commit message to determine the increment type.
+      Supports the following increment types:
+      - patch (build, chore, ci, docs, fix, perf, refactor, revert, style, test)
+      - minor (feat)
+      - Major (any of the above types with an '!' or 'BREAKING CHANGE:' in commit body)
```

Comment:
Interesting ...  it only parses the last commit 🤔 

This means the consumer of this action have to ensure to remember to have to correct keyword in the _merge commit_ (if using merge commits)

I wonder how other tooling does this? e.g. https://semantic-release.gitbook.io/semantic-release

I wonder if this action should parse every commit between "this" commit, and the "current version" tag commit ?

--

I realised the above, because I wanted to rewrite this description to be:

```suggestion
      `conventional_commits` will parse the commit messages to determine the increment digit for a `semver` version (and implies `semver` scheme)
      Supports the following increment types:
      - patch (build, chore, ci, docs, fix, perf, refactor, revert, style, test)
      - minor (feat)
      - major (any of the above types with an '!' or 'BREAKING CHANGE:' in commit body)
```

Reply:
Yeah... guess this isn't as straight forward :sweat_smile: 

Either this would imply that merge commits would have to follow the same convention to correctly increment the version. 

Or like you said, parse every commit since the last version and work out what increment type the version to jump to. That sounds more 'right' to me.

Reply:
Harder to do in bash though 😞